Src/AutoFixture/DataAnnotations/MinAndMaxLengthAttributeRelay.cs

```diff
@@ -156,7 +156,7 @@ public static bool TryGetFromAttributes(object request, out Range range)
             private static Range GetRange(MinLengthAttribute minLengthAttribute, MaxLengthAttribute maxLengthAttribute)
             {
                 var min = minLengthAttribute?.Length ?? 0;
-                var max = maxLengthAttribute?.Length ?? int.MaxValue;
+                var max = maxLengthAttribute?.Length ?? min + 100;
```

Comment:
What's so special about 100? Why not 1, or 15, or 50? Why not 1337? Can we *ask* the context for a positive integer instead?

Reply:
@moodmosaic Here is the issue we are trying to fix by this PR. Currently to calculate the actual array size we already take a _random_ positive number in `[min;int.MaxValue)` range. But the point is that value might be very large and AutoFixture might generate the array for long period of time or even fail with OOM. We need to somehow restrict AutoFixture to not create arrays of crazy sizes.

Personally I also don't like `100` as a constant, as it isn't bound to the min value somehow. Thinking more about this PR I came up with an idea to consider the upper boundary (if not specified explicitly) as `min * 2`. This will allow to keep the comparable order for the actual size (for instance if `min` is 10, then capacity is in `[10;20]` range, while if `min` is 1000, the capacity is in `[1000;2000]` range). This way capacity will not be crazy unless `min` is not set to some really huge value - but in this case it's fine as we are enforced by the lower boundary anyway.

WDYT? 😉

Reply:
It looks like you're kind of hijacking the purpose of a *relay* that's all 😃 The purpose of relays is to *relay* a request for to one or more ISpecimenBuilder instances; in this case, I'd relay the request.

Then, it'd be the responsibility of some particular ISpecimenBuilder to decide what to do next. And MinAndMaxLengthAttributeRelay would keep having just one responsibility, instead of two.

Reply:
Yes, you are completely right! I'll fix that in my subsequent PR and will make it a pure relay.

Anyhow, the responsibility of this class is to define the range, so the change under discussion is still relevant.

Reply:
>Yes, you are completely right! I'll fix that in my subsequent PR and will make it a pure relay.

👍

>the responsibility of this class is to define the range, so the change under discussion is still relevant.

Why not just `byte.MaxValue` or `int16.MinValue` then, instead of the `100` magic number?

Reply:
@moodmosaic Please refer to [this](https://github.com/AutoFixture/AutoFixture/pull/1051#discussion_r198970259) comment. There I suggested to use `min * 2` and described my reasoning about that... I suggest to proceed with `min x2` rather than `100`.

`byte.MaxValue` and `int16.MaxValue` is not a universal constant, as it could always happen that `min` is greater than `byte.Max` or `int16.Max`...

Reply:
>`byte.MaxValue` and `int16.MaxValue` is not a universal constant, as it could always happen that `min` is greater than `byte.Max` or `int16.Max`...

Same with `min * 2`?

---

All I'm trying to say here is that we better be careful and make sure we're *not* putting a lipstick on a pig... Have we examined the [route cause (4.3.0 → 4.4.0)](https://github.com/AutoFixture/AutoFixture/issues/1050#issuecomment-399574161) of this issue?

Reply:
> Same with min * 2?

Nope, as `min * 2` is always greater than `min` 😅

> Have we examined the route cause (4.3.0 → 4.4.0) of this issue?

Yes. The problem is that currently array length is a _random_ number in `[min;int.MaxValue)` range. Often that capacity is __very__ huge and AutoFixture hangs trying to create such a large array.

To fix the issue we should not create huge arrays if `min` boundary allows that. If we pick random length in `[min;min * 2]` (or `[min;min+100]` as advised by @Pvlerick) range, it would perfectly solve the problem.

Reply:
AFAICT, the problem originates from https://github.com/AutoFixture/AutoFixture/pull/1051/?

Reply:
Yeah, that's the commit where the code has been introduced.

Reply:
Then we must at least let @damian-krychowski be aware of this. The change proposed by @Pvlerick makes sense for now, so let's move forward and merge this.

However, this really signals that we need to introduce the notion of *Range*, so that when there isn't an upper bound, you'll get a small *max* which scales constantly, or linearly, or exponentially.

A similar concept is used [here](https://github.com/hedgehogqa/haskell-hedgehog/blob/master/hedgehog/src/Hedgehog/Internal/Range.hs) and [here](https://github.com/hedgehogqa/fsharp-hedgehog/blob/master/src/Hedgehog/Range.fs) in Hedgehog.